src/pfe/file-watcher/server/src/extensions/projectExtensions.ts

```diff
@@ -177,14 +180,17 @@ async function getExtensionProjectHandler(projectInfo: ProjectInfo): Promise<any
                 if (files) {
                     if (files.includes(".sh-extension")) {
                         handler = new ShellExtensionProject(projectInfo.projectType);
-                        await handler.init(projectInfo);
-                        extensionProjectHandlers[key] = handler;
+                    } else if (files.includes(".odo-extension-config.json")) {
```

Comment:
Why are we removing the handler.init from the appsody case (I assume the above case is for Appsody)? Also, do we need to burn that info as part of the PFE code? Should this be defined as part of the extension manifest and load  that info from it?

Reply:
Actually we are not removing the handler.init, we just move the handler.init to the bottom since both Appsody and ODO can share the same code block for extension project init.

The extension manifest file is more about extension metadata (eg. extension name, extension project type, extension detection, etc), but this config file is more about the extension implementation (eg. build log name, app log name, capabilities) that uses for Turbine layer. ODO inherits the same design from Appsody.

Reply:
As discussed, we'll need a second PR to clean up all odo extension code and do it in generic extension way for the project type specific code via the IExtensionProject interface.